src/domain/flow-system/src/entities/shared/batching_rule.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::Duration;
+
+use crate::{FlowResult, FlowTrigger};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct BatchingRule {
+    min_records_awaited: u64,
+    max_batching_interval: Option<Duration>,
+}
+
+impl BatchingRule {
+    pub fn new(min_records_awaited: u64, max_batching_interval: Option<Duration>) -> Self {
+        assert!(min_records_awaited > 0);
+
+        Self {
+            min_records_awaited,
+            max_batching_interval,
+        }
+    }
+
+    #[inline]
+    pub fn min_records_awaited(&self) -> u64 {
+        self.min_records_awaited
+    }
+
+    #[inline]
+    pub fn max_batching_interval(&self) -> &Option<Duration> {
+        &self.max_batching_interval
+    }
+
+    pub fn evaluate<'a, I: Iterator<Item = &'a FlowTrigger>>(
```

Comment:
I doubt we expect triggers to be in any more complex data structures than `Vec<>`, 
so perhaps we can accept `&[FlowTrigger]` instead of an iterator?

Reply:
I originally started from the idea of walking over primary trigger and secondary triggers in events, but later simplified.